external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -274,46 +324,121 @@ fn parse_comma_list_after_start<F, R>(
+                    if at_stop_set {
+                        break;
+                    }
+                }
+            }
+        } else {
             let current_loc = context.tokens.start_loc();
             let loc = make_loc(context.tokens.file_hash(), current_loc, current_loc);
-            return Err(Box::new(diag!(
+            let diag = diag!(
                 Syntax::UnexpectedToken,
                 (loc, format!("Expected {}", item_description))
-            )));
+            );
+            let at_stop_set = skip_comma_list_item(context, start_token, end_token, diag);
+            if at_stop_set {
+                break;
+            }
         }
-        v.push(parse_list_item(context)?);
-        adjust_token(context.tokens, end_token);
-        if match_token(context.tokens, end_token)? {
-            break Ok(v);
+    }
+    if consume_token(context.tokens, end_token).is_err() {
+        let current_loc = context.tokens.start_loc();
+        let loc = make_loc(context.tokens.file_hash(), current_loc, current_loc);
+        let loc2 = make_loc(context.tokens.file_hash(), start_loc, start_loc);
+        context.env.add_diag(diag!(
+            Syntax::UnexpectedToken,
+            (loc, format!("Expected '{}'", end_token)),
+            (loc2, format!("To match this '{}'", start_token)),
+        ));
+    }
+    v
+}
+
+/// Attempts to skip tokens until the end of the item in the comma-separated list (which started
+/// with an already consumed starting token) - looks for a matched ending token or after a comma.
+fn skip_comma_list_item(
+    context: &mut Context,
+    start_token: Tok,
+    end_token: Tok,
+    diag: Diagnostic,
+) -> bool {
+    context.env.add_diag(diag);
+    let mut start_tokens = 0;
+    let mut stop_parsing = false;
+    loop {
+        if context.at_stop_set() {
+            stop_parsing = true;
+            break;
         }
-        if !match_token(context.tokens, Tok::Comma)? {
-            let current_loc = context.tokens.start_loc();
-            let loc = make_loc(context.tokens.file_hash(), current_loc, current_loc);
-            let loc2 = make_loc(context.tokens.file_hash(), start_loc, start_loc);
-            return Err(Box::new(diag!(
-                Syntax::UnexpectedToken,
-                (loc, format!("Expected '{}'", end_token)),
-                (loc2, format!("To match this '{}'", start_token)),
-            )));
+        if context.tokens.at(Tok::Comma) {
+            advance_record_diag(context);
+            if start_tokens == 0 {
+                break;
+            }
+        };
+        if context.tokens.at(start_token) {
+            start_tokens += 1;
+            advance_record_diag(context);
+            continue;
+        };
+        if context.tokens.at(end_token) {
+            if start_tokens == 0 {
+                break;
+            }
+            start_tokens -= 1;
+            advance_record_diag(context);
+            continue;
```

Comment:
So many continues in here makes my head spin a bit. Though it could be the allergy meds lol
Not sure what else to suggest though